[xinnnyeee] iP

xinnnyeee opened the pull request:
# Ekko
## A chatbot that meows. 
**This chatbot is _extremely cute_!**
> Ekko can manage your todos, and also help you remember everything that you need to remember!

Ekko also makes really good food! `#ffffff`

You can run the jar file by command: 
```
java -jar ekko.jar
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

xinnnyeee committed:
l1: echo method added

xinnnyeee committed:
added Todo class

xinnnyeee committed:
shift main function to a separate file

xinnnyeee committed:
added methods to Todo: markDone, isDone, unDo, printDone and toString

xinnnyeee committed:
removed number field from Todo

xinnnyeee committed:
added methods to Todolist: add, printList

xinnnyeee committed:
edited exit method in Ekko

xinnnyeee committed:
created Deadline and Event classes with basic functionalities

xinnnyeee committed:
bug fixing for structure files

xinnnyeee committed:
Added Command Todo

xinnnyeee committed:
bug fixing for existing commands

xinnnyeee committed:
added "Deadline" command

xinnnyeee committed:
added event command and handled index out of bound error

xinnnyeee committed:
added delete command

xinnnyeee committed:
added testing

xinnnyeee committed:
Fix testing issues

xinnnyeee committed:
Add directory and file creation.

xinnnyeee committed:
Implement method updateFile() to each list-modifying method in Todolist.

xinnnyeee committed:
Merge branch 'branch-Level-7'
Implement file save function in Ekko.

xinnnyeee committed:
Convert dates from String format to LocalDateTime.

xinnnyeee committed:
Fix bug in Deadline.java.

xinnnyeee committed:
Add Parser.java nad Storage.java to handle input parsing and file storage respectively.

xinnnyeee committed:
Fix some bugs and add javaDoc to every file.

xinnnyeee committed:
Make the code more readable.

xinnnyeee committed:
Add "Find" command and provide user manual at greeting.

xinnnyeee committed:
Merge branch 'branch-level-9'
Complete level 9 by adding Find command.

xinnnyeee committed:
Add all classes into a package called ekko.

xinnnyeee committed:
Merge branch 'add-gradle-support'
Add Gradle wrapper files.

xinnnyeee committed:
Add TestParser.java to test two methods in Parser.java.

xinnnyeee committed:
Add TestTodo.java to test two methods in Todo.java.

YashvanGH commented on docs/README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke User Guide
+# EKKO User Guide
```
I like the name of your chatbot! 😄

YashvanGH commented on src/test/java/ekko/core/TestParser.java:
```diff
@@ -0,0 +1,41 @@
+
+public class TestParser {
+    @Test
+    public void parseCommandTest(){
```
Please include a space between the method name and the first brace!

YashvanGH commented on src/test/java/ekko/core/TestParser.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    @Test
+    public void parseDateTimeTest(){
```
Please include a space between the method name and the first brace!

YashvanGH commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,9 @@
+unmark 2
+delete 1
+list
+bye
```
Your ```input.txt``` file looks like it tests most of your commands! But could I ask if there are other tests that have yet to be included? e.g. creating invalid ```Todo```, ```Deadline``` and ```Event```.

YashvanGH commented on src/test/java/ekko/core/TestParser.java:
```diff
@@ -0,0 +1,41 @@
+package ekko.core;  //same package as the class being tested
+import org.junit.jupiter.api.Test;
```
Please leave a space between your package statements and your imports!

YashvanGH commented on src/test/java/ekko/core/TestParser.java:
```diff
@@ -0,0 +1,41 @@
+
+    @Test
+    public void parseDateTimeTest(){
+        assertEquals(parseDateTime("12 Jan 2025"),
+                LocalDateTime.of(2025, 1, 12, 23, 59, 59));
+        assertEquals(parseDateTime("02/03/2025"),
+                LocalDateTime.of(2025, 3, 2, 23, 59, 59));
+        assertEquals(parseDateTime("2025-03-02"),
+                LocalDateTime.of(2025, 3, 2, 23, 59, 59));
+        assertEquals(parseDateTime("Jan 12 2025"),
+                LocalDateTime.of(2025, 1, 12, 23, 59, 59));
+        assertEquals(parseDateTime("12 Jan 2025 12"),
+                LocalDateTime.of(2025, 1, 12, 12, 0, 0));
+        assertEquals(parseDateTime("02/03/2025 12:30"),
+                LocalDateTime.of(2025, 3, 2, 12, 30, 0));
+        assertEquals(parseDateTime("2025-03-02 19:24"),
+                LocalDateTime.of(2025, 3, 2, 19, 24, 0));
+        assertEquals(parseDateTime("Jan 12 2025 01:01"),
+                LocalDateTime.of(2025, 1, 12, 1, 1, 0));
```
I like how comprehensive your ```parseDateTimeTest()``` method is! 👍

YashvanGH reviewed (commented):
Overall, LGTM! Your code just has a few nits here and there. They are mainly just coding standard violations.

Note: If you can configure your ```checkstyle.xml``` and your IDE, that should help with identifying the violations easier. ATB!

Eun777 commented on src/test/java/ekko/task/TestTodo.java:
```diff
@@ -0,0 +1,23 @@
+package ekko.task;
+import org.junit.jupiter.api.Test;
```
Would be good to leave a space between package and import statement.

Eun777 commented on src/test/java/ekko/task/TestTodo.java:
```diff
@@ -0,0 +1,23 @@
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestTodo {
```
It might be better to name as TodoTest instead but overall, good job!

Eun777 reviewed (commented):
Great job! Minor style suggestions made.

xinnnyeee committed:
Set up basic GUI files for Ekko.

xinnnyeee committed:
Complete a working GUI for Ekko.

xinnnyeee committed:
Merge branch 'Level-10'
Merge GUI for Ekko.

xinnnyeee committed:
Add the previously unversioned files.

xinnnyeee committed:
Add assertions to test assumptions.

There was no assertions to verify assumptions.
The lack of assertions make debugging process harder.

Assertions should be added to error-prone places like Todolist.java to ensure the correct execution of functions.

xinnnyeee committed:
Add JavaDoc for newly added functions.

xinnnyeee committed:
Shorten getResponse by moving the functionalities to Parser

xinnnyeee committed:
Use stream for Todolist.java.

xinnnyeee committed:
Add more assertions

xinnnyeee committed:
Merge pull request #4 from xinnnyeee/branch-A-Assertions

Add assertions to code to confirm assumptions

xinnnyeee committed:
Merge pull request #3 from xinnnyeee/branch-A-CodeQuality

Shorten getResponse by moving the functionalities to Parser

xinnnyeee committed:
Merge pull request #2 from xinnnyeee/branch-A-Streams

Use stream for Todolist.java

xinnnyeee committed:
Create class Note and NotesCollection

xinnnyeee committed:
Set up file saving and parsing for note keeping

xinnnyeee committed:
Add JavaDoc for the new classes and methods

xinnnyeee committed:
Adjust MANUAL and add README.md to Ekko

xinnnyeee committed:
Use checkstyle make the code  conform to java coding standard

xinnnyeee committed:
Add checkstyle.xml and suppressions.xml to .gitignore

xinnnyeee committed:
Merge branch 'branch-A-CodingStandard'
Fix coding standard with checkstyle

xinnnyeee committed:
Add css files to style the dialogue, and add personality by changing the profile pictures to self-drawn ones.

xinnnyeee committed:
Add more JUnit tests and exception handling.

xinnnyeee committed:
Merge branch 'branch-Better-GUI'
Add better GUI, JUnit testing and error handling.

xinnnyeee committed:
Update README.md

xinnnyeee committed:
Upload Ui.png

xinnnyeee committed:
Edit README.md to make use of JAR file in implementation guide

xinnnyeee committed:
Tweak README.md

xinnnyeee committed:
Change the user guide in README.md

xinnnyeee committed:
Add handling index out of bound after smoke testing

xinnnyeee committed:
Update README.md with examples

xinnnyeee committed:
Update README.md

xinnnyeee committed:
Tweak UI features

xinnnyeee committed:
Fix storage bugs
